kong/api/routes/upstreams.lua

```diff
@@ -40,7 +42,48 @@ return {
     end,
 
     GET = function(self, dao_factory)
-      crud.paginated_set(self, dao_factory.targets)
+      -- by default, just get a paginated list
+      if not self.params.active then
+        return crud.paginated_set(self, dao_factory.targets)
+      end
+
+      self.params.active = nil
```

Comment:
> Generally speaking, I tend to use path parameters when there is an obvious 'hierarchy' in the resource

I this this is most definitely the case here, where there is a hierarchy between the unused and used targets.